packages/panels/src/Http/Middleware/AuthenticateSession.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Filament\Http\Middleware;
+
+use Filament\Facades\Filament;
+use Illuminate\Session\Middleware\AuthenticateSession as Middleware;
+
+class AuthenticateSession extends Middleware
+{
+    protected function redirectTo($request): ?string
+    {
+        return Filament::getLoginUrl();
```

Comment:
What about this?

```suggestion
        return parent::redirectTo() ?? Filament::getLoginUrl();
```

Also have you checked if the correct panel is always used to generate URLs and not just the default panel? I can't remember if the middleware to identify the panel runs before this or not.

Reply:
The reason I did it like this is just to follow how `Authenticate::class` middleware `redirectTo()` method does it. 

I don't think we should call the `parent::redirectTo()`, because if we would do that, then it will give priority to the `Illuminate\Session\Middleware\AuthenticateSession::redirectUsing(...)` method. It is not unthinkable that people have this callback configured for their entire (client-facing) application without thinking about Filament and handling the specific login URL for Filament. So then people trying to access Filament could suddenly get redirected to a whole other login page based on the `redirectUsing()` callback. So I would say as this is a Filament middleware then you don't need the `redirectUsing()` functionality, just configure it via a panel if you want.

The panel is identified before this middleware, because the Panel `getMiddleware()` method *first* returns the `panel:{id}` middleware alias and then all the other middleware provided in `middleware([...])`. I also tested this myself on a non-default panel. So it should be good.